src/north/cli/goldstone/north/cli/portchannel.py

```diff
@@ -113,26 +116,49 @@ def run_conf(session):
 
 
 def show(session, id=None):
+
     if id != None:
         items = session.get_operational(pcxpath(id))
     else:
         items = get_list(session, "operational")
 
     rows = []
     for item in items:
-        rows.append(
-            [
-                item["portchannel-id"],
-                item["state"]["oper-status"].lower(),
-                item["state"]["admin-status"].lower(),
-                ", ".join(natsorted(list(item["state"].get("interface", [])))),
-            ]
-        )
+        if "config" in item and item["config"]["mode"] == "dynamic":
+            rows.append(
+                [
+                    item["portchannel-id"],
+                    item["state"]["oper-status"].lower(),
+                    item["state"]["admin-status"].lower(),
+                    ", ".join(natsorted(list(item["state"].get("interface", [])))),
+                    item["config"]["mode"],
+                ]
+            )
+        elif "config" in item and item["config"]["mode"] == "static":
+            rows.append(
+                [
+                    item["portchannel-id"],
+                    item["state"]["oper-status"].lower(),
+                    item["state"]["admin-status"].lower(),
+                    ", ".join(natsorted(list(item["config"].get("interface", [])))),
```

Comment:
Why do we need to switch using `config` / `state` depending on the port channel mode?

Reply:
sorry that was added while testing and forgot to optimize the code. it has been corrected now.